Build failure: dolphin-emu

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

Simply trying to build `dolphin-emu`, it fails at the end

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/310539126

### Relevant log output

```console
[ 95%] Building CXX object Source/Core/DolphinQt/CMakeFiles/dolphin-emu.dir/DiscordJoinRequestDialog.cpp.o
[ 95%] Building CXX object Source/Core/DolphinQt/CMakeFiles/dolphin-emu.dir/EmulatedUSB/WiiSpeakWindow.cpp.o
[ 95%] Building CXX object Source/Core/DolphinQt/CMakeFiles/dolphin-emu.dir/FIFO/FIFOAnalyzer.cpp.o
[ 95%] Building CXX object Source/Core/DolphinQt/CMakeFiles/dolphin-emu.dir/FIFO/FIFOPlayerWindow.cpp.o
[ 95%] Building CXX object Source/Core/DolphinQt/CMakeFiles/dolphin-emu.dir/GameList/GameList.cpp.o
[ 95%] Building CXX object Source/Core/DolphinQt/CMakeFiles/dolphin-emu.dir/GameList/GameListModel.cpp.o
[ 96%] Building CXX object Source/Core/DolphinQt/CMakeFiles/dolphin-emu.dir/GameList/GameTracker.cpp.o
[ 96%] Building CXX object Source/Core/DolphinQt/CMakeFiles/dolphin-emu.dir/GameList/GridProxyModel.cpp.o
[ 96%] Building CXX object Source/Core/DolphinQt/CMakeFiles/dolphin-emu.dir/GameList/ListProxyModel.cpp.o
[ 96%] Building CXX object Source/Core/DolphinQt/CMakeFiles/dolphin-emu.dir/GCMemcardCreateNewDialog.cpp.o
[ 96%] Building CXX object Source/Core/DolphinQt/CMakeFiles/dolphin-emu.dir/GCMemcardManager.cpp.o
[ 96%] Building CXX object Source/Core/DolphinQt/CMakeFiles/dolphin-emu.dir/Host.cpp.o
[ 96%] Building CXX object Source/Core/DolphinQt/CMakeFiles/dolphin-emu.dir/HotkeyScheduler.cpp.o
[ 96%] Building CXX object Source/Core/DolphinQt/CMakeFiles/dolphin-emu.dir/InfinityBase/InfinityBaseWindow.cpp.o
[ 96%] Building CXX object Source/Core/DolphinQt/CMakeFiles/dolphin-emu.dir/Main.cpp.o
[ 96%] Building CXX object Source/Core/DolphinQt/CMakeFiles/dolphin-emu.dir/MainWindow.cpp.o
[ 96%] Building CXX object Source/Core/DolphinQt/CMakeFiles/dolphin-emu.dir/MenuBar.cpp.o
/build/source/Source/Core/DolphinQt/MainWindow.cpp:34:10: fatal error: qpa/qplatformnativeinterface.h: No such file or directory
   34 | #include <qpa/qplatformnativeinterface.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [Source/Core/DolphinQt/CMakeFiles/dolphin-emu.dir/build.make:2065: Source/Core/DolphinQt/CMakeFiles/dolphin-emu.dir/MainWindow.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:2523: Source/Core/DolphinQt/CMakeFiles/dolphin-emu.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.17.2-zen1, NixOhEss, 25.11 (Xantusia), 25.11.20251019.5e2a59a`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.32.1`
 - nixpkgs: `/nix/store/7agp54mgffm9m1wc1kgmkm37pvy18qhf-source`



### Notify maintainers

@pbsds 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/do/dolphin-emu/package.nix
+++ b/pkgs/by-name/do/dolphin-emu/package.nix
@@ -2,6 +2,7 @@
   lib,
   stdenv,
   fetchFromGitHub,
+  fetchpatch2,
 
   # nativeBuildInputs
   cmake,
@@ -70,6 +71,13 @@ stdenv.mkDerivation (finalAttrs: {
       popd
     '';
   };
+
+  patches = [
+    (fetchpatch2 {
+      url = "https://github.com/dolphin-emu/dolphin/commit/8edef722ce1aae65d5a39faf58753044de48b6e0.patch?full_index=1";
+      hash = "sha256-QEG0p+AzrExWrOxL0qRPa+60GlL0DlLyVBrbG6pGuog=";
+    })
+  ];
 
   strictDeps = true;
 

